### PR TITLE
Stop registering standard guess plugins in StandardPluginModule, instead from the JAR Manifest via Embulk system property "default_guess_plugins"

### PR DESCRIPTION
**Pull request by dmikurube**
This PR stops registering guess plugins in `StandardPluginModule` and `StandardPluginExtension`, then removes those two classes.

The guess plugins are now registered via Embulk system property `default_guess_plugins`, which is set initially from the JAR Manifest.

### CODE REVIEW

**Review by trung-huynh — APPROVED** [review]
LGTM

File: embulk-core/src/main/java/org/embulk/exec/GuessExecutor.java [hunk 1]
```diff
@@ -56,8 +50,7 @@
     }
 
     public static void registerDefaultGuessPluginTo(Binder binder, PluginType type) {
-        Multibinder<PluginType> multibinder = Multibinder.newSetBinder(binder, PluginType.class, ForGuess.class);
-        multibinder.addBinding().toInstance(type);
+        throw new UnsupportedOperationException("GuessExecutor.registerDefaultGuessPluginTo no longer works.");
```
Comment (shroman) [inline embulk-core/src/main/java/org/embulk/exec/GuessExecutor.java:53]: extreme nit :) `no longer works` sounds like it's broken, more accurate wording could be better.

**Review by shroman — APPROVED** [review]
lgtm

File: embulk-core/src/main/java/org/embulk/exec/GuessExecutor.java (hunk 1 above)
Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/exec/GuessExecutor.java]: Thanks! Agreed. Changed the wording to `is no longer available`.

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**